src/core/meshcop/dataset_manager.hpp

```diff
@@ -110,6 +110,70 @@ class DatasetManager : public InstanceLocator
      */
     Error Read(Dataset::Tlvs &aDatasetTlvs) const { return mLocal.Read(aDatasetTlvs); }
 
+    /**
+     * Saves the Operational Dataset in non-volatile memory.
+     *
+     * @param[in]  aDataset  The Operational Dataset.
+     *
+     * @retval kErrorNone   Successfully applied configuration.
+     * @retval kErrorParse  The dataset has at least one TLV with invalid format.
+     *
+     */
+    Error SaveLocal(const Dataset &aDataset);
+
+    /**
+     * Saves the Operational Dataset in non-volatile memory.
+     *
+     * @param[in]  aDatasetInfo  The Operational Dataset as `Dataset::Info`.
+     *
+     * @retval kErrorNone             Successfully saved the dataset.
+     * @retval kErrorNotImplemented   The platform does not implement settings functionality.
```

Comment:
Implementing non-volatile should be a hard requirement.

Reply:
Yes. I noticed that as well. :). For now I copied the existing method documentation. 

Was thinking to change this in a future PR, maybe in more general way in `Settings` for all interactions and items we may `Save` in settings (`assert` in `Settings` class directly rather than passing error to the caller). 

Let me see if we can update this (for `Dataset`) in this PR, if it becomes too big of a change, will leave it for its own future separate PR.

Reply:
I'm fine to have a separate PR.